runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -0,0 +1,499 @@
+				return "Sandstone";
+			case ObjectID.ROCKS_7457: // Silver
+			case ObjectID.ROCKS_7490: // Silver
+				return "Silver";
+			case ObjectID.ROCKS_7456: // Coal
+			case ObjectID.ROCKS_7489: // Coal
+				return "Coal";
+			case ObjectID.ROCKS_7458: // Gold
+			case ObjectID.ROCKS_7491: // Gold
+				return "Gold";
+			case ObjectID.ROCKS_7459: // Mithril
+			case ObjectID.ROCKS_7492: // Mithril
+				return "Mithril";
+			case ObjectID.ROCKS_7460: // Adamant
+			case ObjectID.ROCKS_7493: // Adamant
+				return "Adamantite";
+			case ObjectID.ROCKS_7461: // Rune
+			case ObjectID.ROCKS_7494: // Rune
+				return "Runite";
+			case ORE_VEIN_26661: // Motherlode vein
+			case ORE_VEIN_26662: // Motherlode vein
+			case ORE_VEIN_26663: // Motherlode vein
+			case ORE_VEIN_26664: // Motherlode vein
+			case DEPLETED_VEIN_26665: // Depleted motherlode vein
+			case DEPLETED_VEIN_26666: // Depleted motherlode vein
+			case DEPLETED_VEIN_26667: // Depleted motherlode vein
+			case DEPLETED_VEIN_26668: // Depleted motherlode vein
+				return "Pay-dirt";
+			case CRYSTALS: // Amethyst crystals
+			case CRYSTALS_30372: // Amethyst crystals
+			case EMPTY_WALL: // Depleted amethyst crystals
+				return "Amethyst";
+		}
+		return "";
+	}
+
+	@Subscribe
+	public void onGameObjectDespawned(GameObjectDespawned event)
+	{
+		final GameObject object = event.getGameObject();
+		final Player player = client.getLocalPlayer();
+
+		if (object.getWorldLocation().distanceTo(player.getWorldLocation()) > 20)
+		{
+			return;
+		}
+		objects = removeDuplicates(objects);
+		if (isDepleted(object))
+		{
+			Iterator<Obj> iterator = objects.stream().distinct().iterator();
+			while (iterator.hasNext())
+			{
+				Obj obj = iterator.next();
+				GameObject o = obj.getGameObject();
+				if (object.getX() == o.getX() && object.getY() == o.getY() && object.getPlane() == o.getPlane())
+				{
+					objects.set(objects.indexOf(obj), new Obj(object, System.currentTimeMillis(), 0));
+				}
+			}
+		}
+		else if (isRock(object))
+		{
+			objects.add(new Obj(object, System.currentTimeMillis(), getRespawnTime(object)));
+		}
+	}
+
+	@Subscribe
+	public void onWallObjectSpawned(WallObjectSpawned event)
+	{
+		final WallObject object = event.getWallObject();
+		final Player player = client.getLocalPlayer();
+
+		if (object.getWorldLocation().distanceTo(player.getWorldLocation()) > 20)
+		{
+			return;
+		}
+		if (isDepleted(object))
+		{
+			wallObjects.add(new WallObj(object, System.currentTimeMillis(), getRespawnTime(object)));
+		}
+		else if (isRock(object))
+		{
+			Iterator<WallObj> iterator = wallObjects.stream().distinct().iterator();
+			while (iterator.hasNext())
+			{
+				WallObj obj = iterator.next();
+				WallObject o = obj.getWallObject();
+				if (object.getX() == o.getX() && object.getY() == o.getY() && object.getPlane() == o.getPlane())
+				{
+					wallObjects.set(wallObjects.indexOf(obj), new WallObj(object, System.currentTimeMillis(), 0));
```

Comment:
Why does this timer start both when the wall spawns and despawns?

Reply:
This sets the timer to 0, so it no longer appears on the overlay, as when calling remove(wallObjects.indexOf(obj)) instead, it was causing a lot of issues.

The one below that (else if) is to add the respawn timer to the list.